Create a CI build for 32-bits on Linux

Motivated by #10772.  I think there are multiple little errors we will need to fix.  It may take a while to clean up these errors so I propose we do the following:

- [x] Introduce the CI build, but ignore build and test failures in the driver script (e.g., `cmake ... || true`).
- [x] Fix `google/cloud/storage/tests/object_read_large_integration_test.cc:43`
- [x] Fix `examples/grpc_credential_types.cc:106`
- [x] Fix `examples/grpc_credential_types.cc:135`
- [x] Fix `google/cloud/storage/benchmarks/aggregate_upload_throughput_benchmark.cc:323`
- [x] Fix `google/cloud/spanner/value_test.cc:189`
- [x] Fix `google/cloud/spanner/value_test.cc:763`
- [x] Fix the exceptions thrown in `pubsub_internal_subscription_message_queue_test`
- [x] Fix the spanner integration test freezes (or at least very long delays)
- [x] Fix the Pub/Sub integration tests (`pubsub_endurance`, `pubsub_samples`)
- [x] Change the script to fail when something breaks

```
FAILED: google/cloud/storage/tests/CMakeFiles/storage_object_read_large_integration_test.dir/object_read_large_integration_test.cc.o 
/usr/bin/ccache /usr/bin/g++ -DGOOGLE_CLOUD_CPP_HAVE_GETRUSAGE=1 -DGOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD=1 -DGTEST_LINKED_AS_SHARED_LIBRARY=1 -I/workspace -isystem /workspace/cmake-out -m32 -Wall -Wextra -Wconversion -Wno-sign-conversion -Werror -MD -MT google/cloud/storage/tests/CMakeFiles/storage_object_read_large_integration_test.dir/object_read_large_integration_test.cc.o -MF google/cloud/storage/tests/CMakeFiles/storage_object_read_large_integration_test.dir/object_read_large_integration_test.cc.o.d -o google/cloud/storage/tests/CMakeFiles/storage_object_read_large_integration_test.dir/object_read_large_integration_test.cc.o -c /workspace/google/cloud/storage/tests/object_read_large_integration_test.cc
/workspace/google/cloud/storage/tests/object_read_large_integration_test.cc: In function 'std::size_t google::cloud::storage::v2_8_0::{anonymous}::CurrentRss()':
/workspace/google/cloud/storage/tests/object_read_large_integration_test.cc:43:33: error: conversion from 'long long unsigned int' to 'std::size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   43 |   return std::stoull(fields[1]) * 4096;
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
cc1plus: all warnings being treated as errors
```

```
FAILED: examples/CMakeFiles/grpc_credential_types.dir/grpc_credential_types.cc.o 
/usr/bin/ccache /usr/bin/g++ -DGOOGLE_CLOUD_CPP_HAVE_GETRUSAGE=1 -DGOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD=1 -DGTEST_LINKED_AS_SHARED_LIBRARY=1 -I/workspace -isystem /workspace/cmake-out/examples -isystem /workspace/cmake-out -isystem /workspace/cmake-out/external/googleapis -m32 -Wall -Wextra -Wconversion -Wno-sign-conversion -Werror -MD -MT examples/CMakeFiles/grpc_credential_types.dir/grpc_credential_types.cc.o -MF examples/CMakeFiles/grpc_credential_types.dir/grpc_credential_types.cc.o.d -o examples/CMakeFiles/grpc_credential_types.dir/grpc_credential_types.cc.o -c /workspace/examples/grpc_credential_types.cc
/workspace/examples/grpc_credential_types.cc: In lambda function:
/workspace/examples/grpc_credential_types.cc:106:76: error: conversion from 'int64_t' {aka 'long long int'} to 'time_t' {aka 'long int'} may change value [-Werror=conversion]
  106 |         std::chrono::system_clock::from_time_t(token->expire_time().seconds());
      |                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/workspace/examples/grpc_credential_types.cc: In function 'void {anonymous}::UseAccessTokenUntilExpired(google::cloud::iam::v2_8_0::IAMCredentialsClient, const std::vector<std::__cxx11::basic_string<char> >&)':
/workspace/examples/grpc_credential_types.cc:135:73: error: conversion from 'int64_t' {aka 'long long int'} to 'time_t' {aka 'long int'} may change value [-Werror=conversion]
  135 |       std::chrono::system_clock::from_time_t(token.expire_time().seconds());
      |                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
cc1plus: all warnings being treated as errors
```

```
FAILED: google/cloud/storage/benchmarks/CMakeFiles/storage_aggregate_upload_throughput_benchmark.dir/aggregate_upload_throughput_benchmark.cc.o 
/usr/bin/ccache /usr/bin/g++ -DGOOGLE_CLOUD_CPP_HAVE_GETRUSAGE=1 -DGOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD=1 -DGTEST_LINKED_AS_SHARED_LIBRARY=1 -I/workspace -isystem /workspace/cmake-out -isystem /workspace/cmake-out/external/googleapis -m32 -Wall -Wextra -Wconversion -Wno-sign-conversion -Werror -MD -MT google/cloud/storage/benchmarks/CMakeFiles/storage_aggregate_upload_throughput_benchmark.dir/aggregate_upload_throughput_benchmark.cc.o -MF google/cloud/storage/benchmarks/CMakeFiles/storage_aggregate_upload_throughput_benchmark.dir/aggregate_upload_throughput_benchmark.cc.o.d -o google/cloud/storage/benchmarks/CMakeFiles/storage_aggregate_upload_throughput_benchmark.dir/aggregate_upload_throughput_benchmark.cc.o -c /workspace/google/cloud/storage/benchmarks/aggregate_upload_throughput_benchmark.cc
/workspace/google/cloud/storage/benchmarks/aggregate_upload_throughput_benchmark.cc: In function '{anonymous}::UploadDetail {anonymous}::UploadOneObject(google::cloud::storage::v2_8_0::Client&, const google::cloud::storage_benchmarks::AggregateUploadThroughputOptions&, const UploadItem&, const std::string&, int)':
/workspace/google/cloud/storage/benchmarks/aggregate_upload_throughput_benchmark.cc:323:43: error: conversion from 'long long unsigned int' to 'std::streamsize' {aka 'int'} may change value [-Werror=conversion]
  323 |     if (!stream.write(write_block.data(), n)) break;
      |                                           ^
cc1plus: all warnings being treated as errors
```

```
FAILED: google/cloud/spanner/CMakeFiles/spanner_value_test.dir/value_test.cc.o 
/usr/bin/ccache /usr/bin/g++ -DGOOGLE_CLOUD_CPP_HAVE_GETRUSAGE=1 -DGOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD=1 -DGTEST_LINKED_AS_SHARED_LIBRARY=1 -I/workspace -isystem /workspace/cmake-out -isystem /workspace/cmake-out/external/googleapis -m32 -Wall -Wextra -Wconversion -Wno-sign-conversion -Werror -MD -MT google/cloud/spanner/CMakeFiles/spanner_value_test.dir/value_test.cc.o -MF google/cloud/spanner/CMakeFiles/spanner_value_test.dir/value_test.cc.o.d -o google/cloud/spanner/CMakeFiles/spanner_value_test.dir/value_test.cc.o -c /workspace/google/cloud/spanner/value_test.cc
/workspace/google/cloud/spanner/value_test.cc: In member function 'virtual void google::cloud::spanner::v2_8_0::{anonymous}::Value_BasicSemantics_Test::TestBody()':
/workspace/google/cloud/spanner/value_test.cc:189:8: error: conversion from 'long long int' to 'time_t' {aka 'long int'} may change value [-Werror=conversion]
  189 |        }) {
      |        ^
/workspace/google/cloud/spanner/value_test.cc: In member function 'virtual void google::cloud::spanner::v2_8_0::{anonymous}::Value_ProtoConversionTimestamp_Test::TestBody()':
/workspace/google/cloud/spanner/value_test.cc:763:8: error: conversion from 'long long int' to 'time_t' {aka 'long int'} may change value [-Werror=conversion]
  763 |        }) {
      |        ^
cc1plus: all warnings being treated as errors
```

--- a/ci/cloudbuild/builds/m32.sh
+++ b/ci/cloudbuild/builds/m32.sh
@@ -37,6 +37,4 @@ io::run cmake "${cmake_args[@]}"
 cmake --build cmake-out
 mapfile -t ctest_args < <(ctest::common_args)
 env -C cmake-out ctest "${ctest_args[@]}" -LE "integration-test"
-
-# TODO(#10775) - enable failures on integration tests
-integration::ctest_with_emulators "cmake-out" || true
+integration::ctest_with_emulators "cmake-out"

